### PR TITLE
Feature/sat kabel tap controlling

### PR DESCRIPTION
**Pull request by NinoRy**
#

### CODE REVIEW

File: modules/HfcReq/Http/Controllers/NetElementController.php [hunk 1]
```diff
@@ -30,19 +31,39 @@ public function view_form_fields($netelement = null)
...
-        if (isset($_GET['netelementtype_id'])) {
-            $type = $_GET['netelementtype_id'];
+        if (Request::has('netelementtype_id')) {
+            $type = Request::get('netelementtype_id');
         } elseif ($netelement->netelementtype) {
             $type = $netelement->netelementtype->get_base_type();
         }
 
+        $hidden4TapPort = $hidden4Tap = 0;
+        $hidden4Tap = $type == 8 ? 1 : 0;
```
Comment (olebowle) [inline modules/HfcReq/Http/Controllers/NetElementController.php:47]: Can you move this into the $type == 8 if condition? Like you did for $type == 9.

File: modules/HfcReq/Http/Controllers/NetElementController.php [hunk 2]
```diff
@@ -30,19 +31,39 @@ public function view_form_fields($netelement = null)
...
+        $addressDesc2 = 'Address Line 2';
+
+        if ($type == 8) {
+            $addressDesc1 = 'RKS '.trans('messages.Address'); // Used as address to control the attenuation setting via Sat-Kabel-RKS-Server
+            $addressDesc2 = trans('messages.Address');
+        }
+
+        if ($type == 9) {
+            $parents = $netelement->html_list(NetElement::where('netelementtype_id', 8)->get(), 'name');
+            $types = NetElementType::where('id', 9)->get(['id', 'name']);
```
Comment (olebowle) [inline modules/HfcReq/Http/Controllers/NetElementController.php:58]: I would suggest using where('id', $type) -> single point of truth

File: modules/HfcSnmp/Resources/lang/de/help.php [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<?php
+
+return [
+    'lineNr' => 'Die eingetragene Leitungsnummer wurde über das Cluster ermittelt, in dem sich der Tap-Port befindet. Wenn Sie das Live-Bild eines anderen Clusters sehen möchten, können sie einfach die Nummer ändern.',
+    'rkmServerAddress' => 'IP:Port (IP kann auch der Domain name sein)',
```
Comment (olebowle) [inline modules/HfcSnmp/Resources/lang/de/help.php:5]: Maybe: IP kann auch ein Hostname sein

File: modules/HfcReq/Resources/views/NetElement/tapControlling.blade.php [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+@DivClose()
+
+    <div class="col-md-5 m-l-20">
+        @DivOpen(1)
+        @DivClose()
+        @DivOpen(6)
+            @if (! $hfcBaseConf->video_encoder)
+                <div class="m-t-10 alert alert-danger ajaxResponse">{{ trans('hfcsnmp::messages.missingConf', ['device' => 'Video Encoder']) }}</div>
+            @else
+            <!-- <img id="video" src="http://{{$hfcBaseConf->video_encoder}}/mjpg/1/video.mjpg?camera=1" alt="http://193.168.231.77:2019/mjpg/1/video.mjpg?camera=1"> -->
```
Comment (olebowle) [inline modules/HfcReq/Resources/views/NetElement/tapControlling.blade.php:76]: I think we should remove the actual IP address in alt tag.

**Comment by olebowle** [comment]
Regarding 18389d4f8e6896fcbbb89f98970e9bec0ab19066: Can't we check for uniqueness in prepare_rules()?

File: modules/HfcReq/Resources/views/NetElement/tapControlling.blade.php (hunk 4 above)
Comment (NinoRy) [reply modules/HfcReq/Resources/views/NetElement/tapControlling.blade.php]: Thx! This wouldn't even work with another server

File: modules/HfcSnmp/Resources/lang/de/help.php (hunk 3 above)
Comment (NinoRy) [reply modules/HfcSnmp/Resources/lang/de/help.php]: Thx. Is better!

File: modules/HfcReq/Http/Controllers/NetElementController.php (hunk 2 above)
Comment (NinoRy) [reply modules/HfcReq/Http/Controllers/NetElementController.php]: changed

File: modules/HfcReq/Http/Controllers/NetElementController.php (hunk 1 above)
Comment (NinoRy) [reply modules/HfcReq/Http/Controllers/NetElementController.php]: changed

**Comment by NinoRy** [comment]
> Regarding [18389d4](https://github.com/nmsprime/nmsprime/commit/18389d4f8e6896fcbbb89f98970e9bec0ab19066): Can't we check for uniqueness in prepare_rules()?

**Pull request merged**